### ISSUE TITLE
enable redis as optional feature

### ISSUE THREAD

**Issue by tomfankhaenel** [sse-secure-systems/connaisseur#1587]
**Describe the feature**
Since 3.4.0 the a redis is shipped by default and can not be disabled. 
In my opinion it should be possible to disable the redis completely or at least use an external redis.

**Optional: Is your feature request related to a problem? Please describe.**
The problem is that all cache/db services in my cluster must run cluster external (as a service).

**Optional: Implementation ideas**
Nothing fancy in helm but a few values to enable/disable the feature.

**Comment by Starkteetje** [comment]
Hey @tomfankhaenel 
I think that's a good idea and we actually planned to implement the disabling part, but didn't get to it yet.

As for the external Redis, if we did expose this, we would likely make this config a bit more generic to allow subsequent changes like different key-value stores. Maybe I can pick your brain a bit on what would work for your use case. I would envision something along the lines of

```yaml
application:
  features:
    cache:
      expirySeconds: 30 // <= 0 means disabled
      cacheErrors: true // whether to cache validation errors
      
      type: redis
      url: your.domain.here:6379
      auth:
        // something type-specific here
        user: for-example-user
        password: and-password-combo
```

Then, if `expirySeconds<=0` or `url` is set, we wouldn't deploy the Redis as part of the Connaisseur deployment. Would something be missing from your point of view or do you have any other feedback?

Generally, disabling the Redis should be an easy change. Maybe I'll even be able to get it done today (though not released). Making a remote Redis (or other KVS) configurable will take more time as it'll need more testing, more care with regards to the exposed config API and additionally I think we'll first prioritize our time on improving Cosign support (e.g. keyless) or adding a Notation validator

**Comment by Starkteetje** [comment]
The toggle to enable Redis (and its deployment) has now been released with 3.5.0. You can find the config options in the [docs](https://sse-secure-systems.github.io/connaisseur/latest/features/caching/)
